transport/conn.go

```diff
@@ -325,6 +338,10 @@ func WrapConn(conn net.Conn) (*Conn, error) {
 		},
 		metrics: m,
 	}
+	c.w.reset = func() {
+		c.w.conn.Reset(c.conn)
```

Comment:
Should I choose here the same approach as in [send rcv error PR](https://github.com/mmatczuk/scylla-go-driver/pull/145#discussion_r828505756)?